app/pages/lab/workflows-container.jsx

```diff
@@ -15,17 +16,35 @@ export default class WorkflowsContainer extends React.Component {
     this.hideCreateWorkflow = this.hideCreateWorkflow.bind(this);
     this.showCreateWorkflow = this.showCreateWorkflow.bind(this);
     this.handleWorkflowReorder = this.handleWorkflowReorder.bind(this);
+    this.toggleReorder = this.toggleReorder.bind(this);
+    this.onPageChange = this.onPageChange.bind(this);
   }
 
   componentDidMount() {
     this.getWorkflowList();
   }
 
-  getWorkflowList() {
-    getWorkflowsInOrder(this.props.project, { fields: 'display_name' })
-    .then((workflows) => {
-      this.setState({ workflows, loading: false });
+  onPageChange(page) {
+    const nextQuery = Object.assign({}, this.props.location.query, { page });
+    this.context.router.push({
+      pathname: this.props.location.pathname,
+      query: nextQuery
     });
+    this.getWorkflowList(page);
+  }
+
+  getWorkflowList(page = 1) {
+    if (this.state.reorder) {
+      getWorkflowsInOrder(this.props.project, { fields: 'display_name', page_size: this.props.project.links.workflows.length })
+      .then((workflows) => {
+        this.setState({ workflows, loading: false });
+      });
+    } else {
+      this.props.project.get('workflows', { page })
+        .then((workflows) => {
+          this.setState({ workflows, loading: false });
```

Comment:
clicking Next doesn't advance to page 2 like I'd expect in https://reworked-workflows-ui.pfe-preview.zooniverse.org/lab/1613/workflows. I've been trying to figure out why without luck, the url is properly updated, maybe a `componentWillReceiveProps` that compares `location.query.page`?